sql/logictest: TestParallel failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/6d8bd75a2cf8ee76d5e9a3999e3544d78acaa1e4

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stressrace instead of stress and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
make stress TESTS=TestParallel PKG=github.com/cockroachdb/cockroach/pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-stderr=false -maxtime 20m -timeout 10m'
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=930655&tab=buildLog

```

I180928 05:22:55.685000 941 storage/split_queue.go:116  [n1,split,s1,r5/1:/System/ts{d-e}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685039 941 storage/split_queue.go:116  [n1,split,s1,r9/1:/Table/1{2-3}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685078 941 storage/split_queue.go:116  [n1,split,s1,r4/1:/System/{NodeLive…-tsd}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685110 941 storage/split_queue.go:116  [n1,split,s1,r5/1:/System/ts{d-e}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685149 941 storage/split_queue.go:116  [n1,split,s1,r12/1:/{Table/15-Max}] shouldQueue: shouldQ=true priority=1.0
I180928 05:22:55.685202 941 storage/split_queue.go:116  [n1,split,s1,r11/1:/Table/1{4-5}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685232 941 storage/split_queue.go:116  [n1,split,s1,r1/1:/{Min-System/}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685264 941 storage/split_queue.go:116  [n1,split,s1,r8/1:/Table/1{1-2}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685300 941 storage/split_queue.go:116  [n1,split,s1,r7/1:/Table/{SystemCon…-11}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685330 941 storage/split_queue.go:116  [n1,split,s1,r3/1:/System/NodeLiveness{-Max}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685366 941 storage/split_queue.go:116  [n1,split,s1,r9/1:/Table/1{2-3}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685391 941 storage/split_queue.go:116  [n1,split,s1,r6/1:/{System/tse-Table/System…}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685430 941 storage/split_queue.go:116  [n1,split,s1,r10/1:/Table/1{3-4}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.685455 941 storage/split_queue.go:116  [n1,split,s1,r2/1:/System/{-NodeLive…}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.691506 1033 sql/event_log.go:126  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I180928 05:22:55.694678 741 server/server.go:1606  [n1] done ensuring all necessary migrations have run
I180928 05:22:55.694704 741 server/server.go:1609  [n1] serving sql connections
I180928 05:22:55.695408 941 storage/split_queue.go:116  [n1,split,s1,r4/1:/System/{NodeLive…-tsd}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.695456 941 storage/split_queue.go:116  [n1,split,s1,r3/1:/System/NodeLiveness{-Max}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.695483 941 storage/split_queue.go:116  [n1,split,s1,r2/1:/System/{-NodeLive…}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.695527 941 storage/split_queue.go:116  [n1,split,s1,r1/1:/{Min-System/}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.695570 941 storage/split_queue.go:116  [n1,split,s1,r12/1:/{Table/15-Max}] shouldQueue: shouldQ=true priority=1.0
I180928 05:22:55.695602 941 storage/split_queue.go:116  [n1,split,s1,r8/1:/Table/1{1-2}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.695632 941 storage/split_queue.go:116  [n1,split,s1,r9/1:/Table/1{2-3}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.695665 941 storage/split_queue.go:116  [n1,split,s1,r5/1:/System/ts{d-e}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.696101 941 storage/split_queue.go:116  [n1,split,s1,r13/1:/{Table/16-Max}] shouldQueue: shouldQ=true priority=1.0
I180928 05:22:55.696152 941 storage/split_queue.go:116  [n1,split,s1,r11/1:/Table/1{4-5}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.696210 941 storage/split_queue.go:116  [n1,split,s1,r10/1:/Table/1{3-4}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.696247 941 storage/split_queue.go:116  [n1,split,s1,r7/1:/Table/{SystemCon…-11}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.696277 941 storage/split_queue.go:116  [n1,split,s1,r6/1:/{System/tse-Table/System…}] shouldQueue: shouldQ=false priority=0.0
I180928 05:22:55.696362 892 storage/split_queue.go:116  [n1,split,s1,r12/1:/Table/1{5-6}] shouldQueue: shouldQ=false priority=0.0
```

```
--- FAIL: TestParallel (11.26s)
    --- PASS: TestParallel/subquery_retry (0.89s)
    --- FAIL: TestParallel/subquery_retry_multinode (10.37s)
    	test_server_shim.go:176: had 13 ranges at startup, expected 22
```

--- a/build/teamcity-stress.sh
+++ b/build/teamcity-stress.sh
@@ -46,7 +46,7 @@ go install ./pkg/cmd/github-post
 # Use an `if` so that the `-e` option doesn't stop the script on error.
 if ! make stress PKG="$PKG" TESTTIMEOUT=40m GOFLAGS="$GOFLAGS" TAGS="$TAGS" STRESSFLAGS="-maxruns 100 -maxfails 1 -stderr $STRESSFLAGS" 2>&1 \
   | tee artifacts/stress.log; then
-  exit_status=$?
+  exit_status=${PIPESTATUS[0]}
   go tool test2json -t < artifacts/stress.log | github-post
   exit $exit_status
 fi

